WordPress/gutenberg
# Made SiteLogoReplaceFlow always available in the Site Logo block toolbar

Makes the SiteLogoReplaceFlow always available in the toolbar, to render "Choose Site Logo" when there is no logo, and "Replace" when there is one.  Part of https://github.com/WordPress/gutenberg/issues/63327. 

The "Choose Site Logo" string should map to the same as what we decide for the Inspector, as I explored in https://github.com/WordPress/gutenberg/pull/63498. 

## Why?
It should be clearer how to add a site logo and using an existing affordance that we apply elsewhere is much more supportive of the user experience. 

## Testing Instructions
1. Open a post or page.
2. Insert the Site Logo block.
3. Reset the logo if there is an existing one uploaded already.
3. See "Choose Site Logo" in the toolbar. 
4. Choose an image to use as your logo. 
5. See "Replace" in the toolbar. 
6. Select "Replace" and see a third menu option to "Reset". 


## Screenshots or screencast 

| Before  | After |
| ------------- | ------------- |
|![CleanShot 2024-07-12 at 15 05 17](https://github.com/user-attachments/assets/9d989aaa-ecf0-4938-9e09-ae66702b5583)|![CleanShot 2024-07-22 at 15 46 00](https://github.com/user-attachments/assets/a3755404-44de-4596-8754-d737016c23a0)|

---

Added by @t-hamano 

The `SiteLogoReplaceFlow` component is a wrapper component for `MediaReplaceFlow`, and the Reset button is rendered as a child of `MediaReplaceFlow`. When the Reset button is pressed, it is expected that the popover will be closed and the focus will return to the block toolbar button, but currently `MediaReplaceFlow` cannot call `onClose` from the child.

Therefore, in this PR, we will update `MediaReplaceFlow` to support "function as a child". function as a child can receive `onClose` as a prop.
